pkg/stream/environment.go

```diff
@@ -268,10 +268,23 @@ func (e *Environment) QuerySequence(ctx context.Context, reference, stream strin
 	logger := raw.LoggerFromCtxOrDiscard(ctx)
 	rn := rand.Intn(100)
 	n := len(e.locators)
-
 	var lastError error
+
+	if !validateStringParameter(reference) {
+		lastError = fmt.Errorf("producer reference invalid: %s", reference)
+		logger.Error("invalid producer reference", slog.Any("error", lastError))
+		return uint64(0), lastError
```

Comment:
If we return the error, we should not log it. The caller of this function is going to handle the error, perhaps just by logging it; in that case, we will produce a duplicated log (one from our function, one from the caller). We should either return the error or log it. In this specific case, I'm leaning towards just returning it.

Reply:
Ah yes you are right. I agree that we should just return the error in this case. Resloved here [a731701](https://github.com/rabbitmq/rabbitmq-stream-go-client/pull/230/commits/a731701a0f824ab566dee02fa1284b4ce5f000f0)